crypto/merkle/merkle.proto

```diff
@@ -4,7 +4,7 @@ option go_package = "github.com/tendermint/tendermint/crypto/merkle";
 
 // For more information on gogo.proto, see:
 // https://github.com/gogo/protobuf/blob/master/extensions.md
-import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "third_party/proto/gogoproto/gogo.proto";
```

Comment:
this would be considered breaking as when users fetch these files they will also need to fetch third_party dir. Are we fine with this breakage in a minor release @melekes @tessr you can see how it changes things here: https://github.com/tendermint/rust-abci/pull/116

Reply:
it only breaks developer experience & our tutorials (java, kotlin) which rely on protobuf, correct?

Reply:
yes that is the only thing it changes. To merge this I will add a section to the upgrading.md file and see about updating the tutorials.